src/Glennsl__BsBenchmarkJs.re

```diff
@@ -0,0 +1,971 @@
+
+    [@bs.send.pipe: t] external run: t = "run";
+    [@bs.send.pipe: t] external runWithOptions: options => t = "run";
+
+    [@bs.send.pipe: t] external clone: t => t = "clone";
+    [@bs.send.pipe: t] external cloneWithOptions: (t, options) => t = "clone";
+
+  };
+
+  let make: (~options: options=?, string, testFn) => t = (~options=?, name, fn) =>
+    switch (options) {
+    | (None) => Internal.make(name, fn)
+    | (Some(opt)) => Internal.makeWithOptions(name, fn, opt)
+    };
+
+  let run: (~options: options=?, t) => t = (~options=?, benchmark) =>
+    switch (options) {
+    | None => Internal.run(benchmark)
+    | Some(opt) => Internal.runWithOptions(opt, benchmark)
+    };
+
+  let clone: (~options: options=?, t, t) => t =
+    (~options=?, benchmark) =>
+      switch (options) {
+      | None => Internal.clone(benchmark)
+      | Some(opt) => Internal.cloneWithOptions(benchmark, opt)
+      };
+
+  [@bs.get] external getAborted: t => bool = "aborted";
+
+  [@bs.get] external getCompiled: t => testFn = "compiled";
+  
+  [@bs.get] external getCycles: t => int = "cycles";
+  
+  [@bs.get] external getCount: t => int = "count";
+  
+  [@bs.get] external getError: t => Js.Null_undefined.t(Js.Exn.t) = "error";
+  
+  [@bs.get] external getFn: t => testFn = "fn";
+  
+  [@bs.get] external getHz: t => float = "hz";
+  
+  [@bs.get] external getRunning: t => bool = "running";
+  
+  [@bs.get] external getSetup: t => setupFn = "setup";
+  
+  [@bs.get] external getStats: t => stats = "stats";
+  
+  [@bs.get] external getTeardown: t => teardownFn = "teardown";
+  
+  [@bs.get] external getTime: t => times = "time";
+  
+  [@bs.get] external getName: t => string = "name";
+  
+  [@bs.get] external getOptions: t => options = "options";
+      
+  [@bs.send.pipe: t] external abort: t = "abort";
+  
+  [@bs.send.pipe: t] external toString: string = "toString";
+  
+  [@bs.send.pipe: t] external cmp: t => int = "compare";
+  
+  [@bs.send.pipe: t] external reset: t = "reset";
+  
+};
+
+module Deferred = {
+  
+  type t = deferred;
+  
+  [@bs.get] external getBenchmark: t => benchmark = "benchmark";
+
+  [@bs.get] external getCycle: t => float = "cycle";
+  
+  [@bs.get] external getElapsed: t => float = "elapsed";
+  
+  [@bs.get] external getTimeStamp: t => float = "timeStamp";
+  
+};
+
+module Event = {
+
+  type t = event;
+
+  [@bs.module "benchmark"] [@bs.scope "Benchmark"] [@bs.new] external make: string => t = "Event";
+  
+  [@bs.module "benchmark"] [@bs.scope "Benchmark"] [@bs.new] external fromEvent: t => t = "Event";
+  
+  [@bs.get] external getAborted: t => bool = "aborted";
+
+  [@bs.get] external getCancelled: t => bool = "cancelled";
+  
+  [@bs.get] external getCurrentTarget: t => benchmark = "currentTarget";
+  
+  [@bs.get] external getResult: t => 'a = "result";
+  
+  [@bs.get] external getTarget: t => benchmark = "target";
+  
+  [@bs.get] external getTimeStamp: t => int = "timeStamp";
```

Comment:
I initially thought it was an `int`, since it's an integer value of milliseconds since Unix Epoch. But that's obviously to large for a 32-bit integer. I suppose it's a 64-bit `float`, since that's the default "number" type in JS. So I'll fix that.

Reply:
It's represented as a float in BuckleScript's `Js.Date` too, so that's a good choice I think.